### PR TITLE
Better login for Cucumber

### LINKED ISSUE #112: Remove the need for "create and log in a user" backgrounds to all scenarios
I think it'd be better to handle that with a Before hook, and then have a tag to allow scenarios to bypass the login hook.

### PR DESCRIPTION
**Pull request by gjvis**
Closes #112

### CODE REVIEW

File: features/support/authentication.rb [hunk 1]
```diff
@@ -1,9 +1,11 @@
-Given(/^a new, authenticated user$/) do
-  email = 'guider@example.com'
-  password = 'pensionwise'
+email = 'guider@example.com'
+password = 'pensionwise'
 
+unless User.exists?(email: email)
   User.create(email: email, password: password, password_confirmation: password).confirm!
```
Comment (stevenwilkin) [inline features/support/authentication.rb]: Could potentially use [User#first_or_create](http://guides.rubyonrails.org/v3.2.17/active_record_querying.html#first_or_create)?

Comment (gjvis) [inline features/support/authentication.rb]: Definitely. Will change it.

**Comment by stevenwilkin** [comment]
Looks good, I like that we're removing the repeated background step in all those features

**Comment by gjvis** [comment]
This now includes an updated RSpec spec as well

**Pull request merged**